### ISSUE TITLE
[Feature Request] jsonPatches/stratetegicMergePatches should support name prefixes

### ISSUE THREAD

**Issue by twiechert** [roboll/helmfile#1386]
Imagine the case where you wanted to add an annotation to all configmaps of a given chart sharing a mutual name prefix.

It would be desirable if this case was supported directly through `jsonPatches` or `stratetegicMergePatches`. From my understanding, there is currently no way around the explicit use of kustomize for this.

**Comment by mumoshu** [comment]
@twiechert Seems like kustomize(which is used internally for patching) allows using regex patterns in names, as documented in https://kubernetes-sigs.github.io/kustomize/api-reference/kustomization/patches/.

Would you mind trying it, and report back if it works?

**Comment by twiechert** [comment]
@mumoshu 

yeah, in fact, I was also referring to kustomize. However, I did not find an example on how to include and apply kustomizations alongside existing charts. 

I believe this goes into the same direction and would be very helpful in this regard https://github.com/roboll/helmfile/issues/1340
